stretch-assignment/index.js

```diff
@@ -0,0 +1,207 @@
+let i = -1;
+let click;
+
+let red = document.querySelector('.block--red');
+let blue = document.querySelector('.block--blue');
+let green = document.querySelector('.block--green');
+let pink = document.querySelector('.block--pink');
+let gray = document.querySelector('.block--gray');
+
+function stop() {
+    clearInterval(click);
+}
+
+
+// RED --------------------------------------------------------------------------------------
```

Comment:
I played around with it, and it's working as intended, except whenever you click on it, the blocks will start acting really weird. Regardless, It's awesome to see how you try to accomplish the stretch assignment so don't let my words discourage you or it seem I'm looking for perfection. 👍

Also, conceptually, you can see these multiple functions can be handled by a callback function. Food for thought.

I recommend looking at Samuel Machat code for alternative ways to accomplishing this.
https://axolotl.github.io/DOM-II/stretch-assignment/index.html